openfisca_france/model/prelevements_obligatoires/impot_revenu/ir.py

```diff
@@ -597,13 +625,16 @@ def function_20020101_20041231(self, simulation, period):
         F1 = f2ca / den * f2dc_bis  # f12
         # Revenus de capitaux mobiliers nets de frais, ouvrant droit à abattement
         # partie négative (à déduire des autres revenus nets de frais d'abattements
-        g12a = -min_(f2dc_bis * (1 - rvcm.abatmob_taux) - F1, 0)
+        g12a = -min_(f2dc_bis * (1 - rvcm.taux_abattement_capitaux_mobiliers) - F1, 0)
         # partie positive
-        g12b = max_(f2dc_bis * (1 - rvcm.abatmob_taux) - F1, 0)
-        rev = g12b + f2gr + f2fu * (1 - rvcm.abatmob_taux)
+        g12b = max_(f2dc_bis * (1 - rvcm.taux_abattement_capitaux_mobiliers) - F1, 0)
+        rev = g12b + f2gr + f2fu * (1 - rvcm.taux_abattement_capitaux_mobiliers)
 
         # Abattements, limité au revenu
-        h12 = rvcm.abatmob * (1 + maries_ou_pacses)
+        if period.start.date >= datetime.date(2012, 1, 01):
+            h12 = 0
+        else:
+            h12 = rvcm.abatmob * (1 + maries_ou_pacses)
```

Comment:
If we do need to do the case disjunction in the formula file, I think we would benefit from a friendly well-named helper function. This would avoid to import `datetime` in the file.